### DISCUSSION TITLE
webcontrol not working

### DISCUSSION THREAD

**Discussion by Piotr-23** [Motion-Project/motion#1811] [General]
Hi, I can't change text_left dynamically. 

What works:
1. curl --user piotr:***** "http://127.0.0.1:8090/0/config/set?text_left=Out:%2033.33"
2. curl -u piotr:***** http://localhost:8090/0/config/list
text_left = Out: 33.33
So, it appears to be working. However, I see no "Out: 33.33" at the bottom left of the image or video...

4. main motion.conf:

```
# Text to be overlayed in the lower left corner of images
text_left fill_it_with_temp
############################################################
# Webcontrol configuration parameters
############################################################
# Port number used for the webcontrol.
webcontrol_port 8090
# Enable IPv6 addresses.
webcontrol_ipv6 off
# Restrict webcontrol connections to the localhost.
webcontrol_localhost off
# Type of configuration options to allow via the webcontrol.
webcontrol_parms 2
# Method that webcontrol should use for interface with user.
webcontrol_interface 1
# The authentication method for the webcontrol
webcontrol_auth_method 1
# Authentication string for the webcontrol. Syntax username:password
webcontrol_authentication piotr:*****
# Use ssl / tls for the webcontrol
webcontrol_tls off
# Full path and file name of the certificate file for tls
; webcontrol_cert value
# Full path and file name of the key file for tls
; webcontrol_key value
# The cross-origin resource sharing (CORS) header for webcontrol
; webcontrol_cors_header value

##############################################################
# Camera config files - One for each camera.
##############################################################
camera /home/piotr/_motion/camera_1_hik_e0.conf
camera /home/piotr/_motion/camera_2_hik_0d.conf
```
These cameras configs have no webcontrol declarations

And I still see "fill_it_with_temp" instead of "Out: 33.33" at the bottom left of the image or video... 

Interestingly, 
/usr/bin/wget --user=piotr --password=***** --delete-after http://localhost:8090/0/action/snapshot
works perfectly....

Please help | P

**Comment by Mr-Dave** [comment]
Probably need to make sure to include the device id  for each camera in the URL.

**Reply by Piotr-23** [reply]
> Probably need to make sure to include the device id for each camera in the URL.

How to do that?

**Comment by tosiara** [comment]
Since you are using config-per-camera, your camera's IDs are numbered:

1 - camera_1_hik_e0.conf
2 - camera_2_hik_0d.conf

0 is used by main config

So try this: `curl --user piotr:***** "http://127.0.0.1:8090/1/config/set?text_left=Out:%2033.33"`

**Reply by Piotr-23** [reply]
> Since you are using config-per-camera, your camera's IDs are numbered:
> 
> 1 - camera_1_hik_e0.conf 2 - camera_2_hik_0d.conf
> 
> 0 is used by main config
> 
> So try this: `curl --user piotr:***** "http://127.0.0.1:8090/1/config/set?text_left=Out:%2033.33"`

Thank you! That worked. Problem solved | P
